internal/service/dataset/iterator.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2022 MIMIRO AS
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dataset
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+
+	"github.com/dgraph-io/badger/v3"
+
+	"github.com/mimiro-io/datahub/internal/service/store"
+)
+
+type IterableDataset interface {
+	At(since uint64) (Iterator, error)
+}
+
+type IterableBadgerDataset struct {
+	db        *badger.DB
+	datasetID uint32
+}
+type Iterator interface {
+	io.Closer
+	Inverse() Iterator
```

Comment:
Not sure of this should be a property on the Iterator instance instead of it's own method. Or even a ReverseIterator implementation. So by calling a factory method table.GetIterator(reverse:true/false) would then just return the correct Iterator (a Next on a ReverseIterator would then go backwards)